### PR TITLE
Add fuzzy prefixing tree walking

### PR DESCRIPTION
**Pull request by mish15**

### CODE REVIEW

File: node.go [hunk 1]
```diff
@@ -242,6 +248,30 @@ func (n *Node) WalkPrefix(prefix []byte, fn WalkFn) {
...
+			distance, ok := levenshtein.HasPrefix(k, prefix, depth)
+			if !ok {
+				return false
+			}
+			v = dfn(v, distance)
+			return fn(k, v)
+		})
+	}
+}
+
```
Comment (banks) [inline node.go]: The code looks good to me at a glance but test cases to show the behaviour and edge cases seem important!

**Comment by banks** [comment]
This is a really interesting idea! I’m impressed with how little code it needed but have a few questions from a quick look through:

## How well does it work? 

It looks like it will do a fuzzy match, but fuzzy matches in a non-trivial corpus quickly lead to a huge amount of noise which then needs at the very least a basic scoring mechanism to make sure the best (by some definition) are shown first. 

As far as I can see (please correct if wrong) there is no concept of scoring here so if you fuzzy match with distance threshold 3 the results for “rabix” would deliver “racer” before “radix”. 

It looks like since DistFn is pluggable and can modify the value returned you can custom score by emitting for example a tuple (value, score) and then the WalkFn could collect those into a container and sort by score. Is this the intended use? A doc example showing that would be a great addition.

File: node.go [hunk 2]
```diff
@@ -3,13 +3,19 @@ package iradix
 import (
 	"bytes"
 	"sort"
+
+	"code.sajari.com/go-immutable-radix/pkg/levenshtein"
```
[the remark this answers is not in the document]
Comment (mish15) [reply node.go]:
```suggestion
	"github.com/hashicorp/go-immutable-radix/pkg/levenshtein"
```

Yeah this is my bad. I'd had a few beers and sent the PR a bit prematurely by accident.
